bindings/go/oci/repository.go

```diff
@@ -73,10 +74,19 @@ type Repository struct {
 
 	// referrerTrackingPolicy defines how OCI referrers are used to track component versions.
 	referrerTrackingPolicy ReferrerTrackingPolicy
+
+	// logger is the logger used for OCI operations.
+	logger *slog.Logger
+}
+
+// WithLogger returns a new context with the repository's logger.
+func (repo *Repository) withLogger(ctx context.Context) context.Context {
+	return slogcontext.With(ctx, repo.logger)
```

Comment:
why dont you just construct this while building the Repository?

Reply:
Good point. I could just pass in logCtx. That's what you mean, right?

Reply:
Yap. I'll do that.